### PR TITLE
Resolves issue #1039 - do_check() documentation

### PR DESCRIPTION
**Pull request by bscott**
Adds `do_check()` callback documentation.

### CODE REVIEW

**Comment by bscott** [comment]
r+

**Comment by metadave** [comment]
you'll need an @thesentinals before the r+

**Comment by smith** [comment]
And it's not working right now. See https://habitat-sh.slack.com/archives/general/p1468417818008298

File: www/source/docs/reference/plan-syntax.html.md [hunk 1]
```diff
@@ -255,6 +255,10 @@ do_download()
 do_verify()
 : The default implementation tries to verify the checksum specified in the plan against the computed checksum after downloading the source tarball to disk. If the specified checksum doesn't match the computed checksum, then an error and a message specifying the mismatch will be printed to stderr. You should not need to override this behavior unless your package does not download any files.
 
+do_check()
+: The default implementation runs post-compile and checks provided two conditions
```
Comment (smith) [inline www/source/docs/reference/plan-syntax.html.md]: I don't think this is clear. The default implementation does nothing, and you're expected to put something like `make test`, or whatever runs your plan's tests after it's built, in it.

These listed in the order they are run, so `do_check` should go between `do_build` and `do_install`.

**Comment by bscott** [comment]
@smith thanks for the feedback

**Comment by smith** [comment]
Looks great!

giphy

**Pull request merged**